dcos/subcommand.py

```diff
@@ -320,6 +382,46 @@ def _find_virtualenv(bin_directory):
     return virtualenv_path
 
 
+def _install_with_binary(
+        package_name,
+        env_directory,
+        binary_cli):
+    """
+    :param package_name: the name of the package
+    :type package_name: str
+    :param env_directory: the path to the directory in which to install the
+                          package's virtual env
+    :type env_directory: str
+    :param binary_cli: binary cli to install
+    :type binary_cli: str
+    :rtype: None
+    """
+
+    # Do not replace util.temptext NamedTemporaryFile
+    # otherwise bad things will happen on Windows
+    with util.temptext() as text_file:
+        fd, requirement_path = text_file
+
+        try:
```

Comment:
Ignoring this part as I think you are going to change it to take into account both `executable` and `zip` files.